adb/android_pubkey.py

```diff
@@ -0,0 +1,157 @@
+"""This file implements encoding and decoding logic for Android's custom RSA
+public key binary format. Public keys are stored as a sequence of
+little-endian 32 bit words. Note that Android only supports little-endian
+processors, so we don't do any byte order conversions when parsing the binary
+struct.
+
+Structure from:
+https://github.com/aosp-mirror/platform_system_core/blob/c55fab4a59cfa461857c6a61d8a0f1ae4591900c/libcrypto_utils/android_pubkey.c
+
+typedef struct RSAPublicKey {
+    // Modulus length. This must be ANDROID_PUBKEY_MODULUS_SIZE.
+    uint32_t modulus_size_words;
+
+    // Precomputed montgomery parameter: -1 / n[0] mod 2^32
+    uint32_t n0inv;
+
+    // RSA modulus as a little-endian array.
+    uint8_t modulus[ANDROID_PUBKEY_MODULUS_SIZE];
+
+    // Montgomery parameter R^2 as a little-endian array of little-endian words.
+    uint8_t rr[ANDROID_PUBKEY_MODULUS_SIZE];
+
+    // RSA modulus: 3 or 65537
+    uint32_t exponent;
+} RSAPublicKey;"""
+
+
+from __future__ import print_function
+
+import os
+import six
+import base64
+import socket
+import struct
+
+import Crypto.Util
+import Crypto.PublicKey.RSA
+
+
```

Comment:
Use a function for size translation:
```suggestion
def size_in(x, s):
    """
    Get the amount of s-sized blocks required to hold x.
    @param x: original size
    @type x: int
    @param s: desired block size
    @type s: int
    @rtype: int
    """
    return ((x - 1) // s + 1)
```